mill-scalablytyped/src/com/github/lolgab/mill/scalablytyped/ScalablyTyped.scala

```diff
@@ -12,14 +12,18 @@ trait ScalablyTyped extends ScalaJSModule {
     ScalablyTypedWorkerApi.scalablyTypedWorker().impl(classpath)
   }
 
-  private def packageJsonSource = Task.Source {
-    scalablyTypedBasePath / "package.json"
-    // Task.workspace / "package.json"
+  private def packageJsonSource = Task.Anon {
+    Task.Source {
+      scalablyTypedBasePath() / "package.json"
+      // Task.workspace / "package.json"
+    }
   }
 
   /** The base path where package.json and node_modules are.
     */
-  def scalablyTypedBasePath: os.Path = os.Path(sys.env("MILL_WORKSPACE_ROOT"))
+  def scalablyTypedBasePath: T[os.Path] = Task {
+    os.Path(sys.env("MILL_WORKSPACE_ROOT"))
+  }
```

Comment:
This needs to be an input task to capture any changes in `MILL_WORKSPACE_ROOT`. Also, you should use `Task.env` instead of `sys.env`, to avoid stale env values. And third, the workspace path can be easily accessed with `mill.api.BuildCtx.workspaceRoot`.